Realm/RLMArray.mm

```diff
@@ -387,7 +435,7 @@ - (id)maxOfProperty:(NSString *)property {
 
 - (id)sumOfProperty:(NSString *)property {
     [self validateAggregateProperty:property method:_cmd allowDate:false];
-    return [_backingArray valueForKeyPath:[@"@sum." stringByAppendingString:property]];
+    return [_backingArray valueForKeyPath:[@"@sum." stringByAppendingString:property]] ?: @0;
```

Comment:
We decided not to match Foundation's behavior because Foundation is wrong; the sum of an empty set isn't undefined.